api/controllers/UserController.js

```diff
@@ -4,93 +4,13 @@
  * @description :: Server-side logic for managing users
  * @help        :: See http://links.sailsjs.org/docs/controllers
  */
-var crypto = require("crypto");
-var https = require("https");
 
-var clientID = "client_id=" + process.env.CLIENT_ID;
-var clientSecret = "&client_secret=" + process.env.CLIENT_SECRET;
-var url = "https://github.com/login/oauth/authorize?";
-var state;
+var State = require("../services/stateService.js");
+var stateService = State.stateService();
 
-var getRandomString = function() {
-  var currentDate = (new Date()).valueOf().toString();
-  var random = Math.random().toString();
-  var hashed = crypto.createHash("sha1").update(currentDate + random).digest("hex");
-  return hashed;
-};
-
-// Makes post request to github to get accessToken
-// Calls checkUsers
-var authenticate = function(req, res, accessCode) {
-  var postOptions = {
-    host: "github.com",
-    path: "/login/oauth/access_token",
-    method: "POST",
-    headers: {
-      "accept": "application/json"
-    }
-  };
-
-  var request = https.request(postOptions, function(response) {
-    response.setEncoding("utf8");
-    var data = "";
-    response.on("data", function(chunk) {
-      data += chunk;
-    });
-    response.on("end", function() {
-      var accessToken = JSON.parse(data).access_token;
-      checkUsers(req, res, accessToken);
-    });
-  });
-
-  request.write(clientID + clientSecret + "&code=" + accessCode);
-  request.end();
-};
-
-// Makes get request to github to get user information
-// Sets req.session.authenticate
-var checkUsers = function(req, res, token) {
-  var getOptions = {
-    host: "api.github.com",
-    path: "/user?access_token=" + token,
-    method: "GET",
-    headers: {
-      "user-agent": "lucere"
-    }
-  };
-
-  var request2 = https.request(getOptions, function(answer) {
-    var userInfo = "";
-    answer.on("data", function(chunk) {
-      userInfo += chunk;
-    });
-    answer.on("end", function() {
-      var ghUserName = JSON.parse(userInfo).login;
+var Auth = require("../services/authService.js");
```

Comment:
Similar to comment above about State...could save a variable here by just using var authService = require(...).authService();